transports/ftp/src/test/java/org/mule/transport/ftp/FtpReconnectionStrategyTestCase.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+
+package org.mule.transport.ftp;
+
+import static java.lang.Integer.parseInt;
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.fail;
+import static org.mockito.Mockito.RETURNS_DEEP_STUBS;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+import org.mule.api.construct.FlowConstruct;
+import org.mule.api.endpoint.EndpointURI;
+import org.mule.api.endpoint.InboundEndpoint;
+import org.mule.api.transport.Connector;
+import org.mule.tck.junit4.FunctionalTestCase;
+import org.mule.tck.junit4.rule.DynamicPort;
+
+public class FtpReconnectionStrategyTestCase extends FunctionalTestCase
+{
+
+    @Rule
+    public DynamicPort ftpPort = new DynamicPort("port");
+
+    private final FlowConstruct flowConstruct = mock (FlowConstruct.class);
+    private final InboundEndpoint inboundEndpoint = mock(InboundEndpoint.class, RETURNS_DEEP_STUBS);
+    private  FtpMessageReceiver ftpMessageReceiver = null;
+    private Connector connector = null;
+    private static final String ASYNCHRONOUS_RECCONECTION_ERROR_MESSAGE = "FTP Connector doesn't support asynchronous retry policies.";
```

Comment:
The point about defining a constant for this was to use it both in the production code and the test